dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/utils/QueryLogDateUtil.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.utils;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import javax.annotation.Nonnull;
+
+public class QueryLogDateUtil {
```

Comment:
1. It is not a utility class as the name suggests, it's a shared state class, so please rename it.
2. The state is shared between tasks, so it should be thread-safe (if this is the design we need to follow).
3. Can't we avoid a shared state here and implement the whole change in a "functional" way. Each task returns a summary that can contain the first and the last entry. At the end, we just need to find MIN and MAX from all summaries.